iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,37 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  AllTypesMatch<["value", "result"]>
+  ]> {
+  let summary = [{returns the (value + (alignment - 1)) & ~(alignment - 1)}];
+  let description = [{
+    Aligns |value| up to the given power-of-two |alignment| if required.
+    https://en.wikipedia.org/wiki/Data_structure_alignment#Computing_padding
+
+    Usage:
+      %0 = util.align $value, $alignment : index
```

Comment:
Usage comment for an op like this seems redundant with the assembly format. Omit?
I could see a usage comment being useful for an op that works together with other ops, but this one is standalone.